components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,365 @@
+    find_parser.add_argument(
+        BEGIN_TS_ARG,
+        dest="begin_ts",
+        type=int,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    find_parser.add_argument(
+        END_TS_ARG,
+        dest="end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    # Delete options
+    del_parser.add_argument(
+        DRY_RUN_ARG,
+        dest="dry_run",
+        action="store_true",
+        help="Preview delete without making changes. Lists errors and files to be deleted.",
+    )
+
+    # Delete subcommands
+    del_subparsers = del_parser.add_subparsers(
+        dest="del_subcommand",
+        required=True,
+    )
+    del_id_parser = del_subparsers.add_parser(
+        BY_IDS_COMMAND,
+        help="Delete archives by ID.",
+    )
+    del_filter_parser = del_subparsers.add_parser(
+        BY_FILTER_COMMAND,
+        help="delete archives within time frame.",
+    )
+
+    # Delete by ID arguments
+    del_id_parser.add_argument(
+        "ids",
+        nargs="+",
+        help="List of archive IDs to delete",
+    )
+
+    # Delete by filter arguments
+    del_filter_parser.add_argument(
+        "begin_ts",
+        type=int,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    del_filter_parser.add_argument(
+        "end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    # Validate and load config file
+    config_file_path = Path(parsed_args.config)
+    try:
+        clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
+        clp_config.validate_logs_dir()
+    except:
+        logger.exception("Failed to load config.")
+        return -1
+
+    database_config = clp_config.database
+    archives_dir = clp_config.archive_output.get_directory()
+    if not archives_dir.exists():
+        logger.error("`archive_output.directory` doesn't exist.")
+        return -1
+
+    if FIND_COMMAND == parsed_args.subcommand:
+        return _find_archives(
+            archives_dir,
+            database_config,
+            parsed_args.begin_ts,
+            parsed_args.end_ts,
+        )
+    elif DEL_COMMAND == parsed_args.subcommand:
+        if BY_IDS_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_ids(
+                archives_dir,
+                database_config,
+                parsed_args.ids,
+                parsed_args.dry_run,
+            )
+        elif BY_FILTER_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_filter(
+                archives_dir,
+                database_config,
+                parsed_args.begin_ts,
+                parsed_args.end_ts,
+                parsed_args.dry_run,
+            )
+
+
+def _find_archives(
```

Comment:
I am thinking, maybe we should also check if the found ID exists on disk? 
if there's a discrepancy (i.e. some archvie is in database but not on the disk), we can print a warning.

Note it won't catch the case where an archive is on disk but not available in the database. but let's not worry about this case first.